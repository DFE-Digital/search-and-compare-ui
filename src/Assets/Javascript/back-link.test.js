import BackLink from "./back-link"

window.addEventListener = jest.fn().mockImplementation((_, cb) => cb())

describe("Back Link module", () => {
  it("exposes a function", () => {
    expect(typeof BackLink).toBe("function")
  })

  describe("init()", () => {
    describe("with no element", () => {
      beforeEach(() => {
        const backLink = new BackLink()
        backLink.init()
      })

      it("does nothing", () => {
        expect(window.addEventListener).not.toHaveBeenCalled()
      })
    })

    describe("with an element", () => {
      let backLink

      beforeAll(() => {
        backLink = new BackLink({})
        backLink.init()
      })

      it("sets up event listener on window", () => {
        expect(window.addEventListener).toHaveBeenCalledWith("load", expect.any(Function))
      })

      describe("when on a page which does have an internal referrer", () => {
        const domElement = { appendChild: jest.fn() }

        beforeAll(() => {
          Object.defineProperty(document, "referrer", { value: "http://localhost", configurable: true })
          new BackLink(domElement).init()
        })

        it("calls appendChild on element with correct arguments", () => {
          expect(domElement.appendChild.mock.calls).toMatchSnapshot()
        })
      })

      describe("when on a page which does NOT have an internal referrer", () => {
        const domElement = { appendChild: jest.fn() }

        beforeAll(() => {
          Object.defineProperty(document, "referrer", { value: "http://google.com", configurable: true })
          new BackLink(domElement).init()
        })

        it("does NOT call appendChild on element", () => {
          expect(domElement.appendChild).not.toHaveBeenCalled()
        })
      })
    })
  })
})
