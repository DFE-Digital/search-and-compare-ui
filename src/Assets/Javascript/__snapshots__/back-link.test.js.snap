// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Back Link module init() with an element when on a page which does have an internal referrer calls appendChild on element with correct arguments 1`] = `
Array [
  Array [
    <a
      class="govuk-back-link"
      href="http://localhost"
    >
      Back to search results
    </a>,
  ],
]
`;
